Main.java

```diff
@@ -1,237 +1,448 @@
-		
+
 		GridPane.setConstraints(transferButton, 2, 2);
-		
+
 		// QUERIES
 		VBox queries = new VBox();
 		queries.setAlignment(Pos.CENTER);
 		// QUERY BY NAME
 		HBox nameQuery = new HBox();
 		TextField nameText = new TextField();
+		nameText.setMinWidth(500);
 		nameText.setPromptText("Filter by Name");
 		Button nameQueryButton = new Button("Add Name Filter");
 		nameQueryButton.setOnAction(e -> {
-			//TODO limit what shows in foodList
+			nameQuery();
+
 		});
 		nameQuery.getChildren().addAll(nameText, nameQueryButton);
-		
+
 		Text orText = new Text("- OR - ");
-		
-		
+
+
 		// QUERY BY NUTRIENTS
 		HBox nutriQuery = new HBox();
-		ComboBox<FoodItem> nutriDrop = new ComboBox<>();
-		ComboBox<FoodItem> compDrop = new ComboBox<>();
-		TextField queryAmount = new TextField();
-		queryAmount.setPromptText("Amount");
+		TextField queryRules = new TextField();
+		queryRules.setMinWidth(500);
+		queryRules.setPromptText("Nutrient Filter Rules");
 		Button nutriQueryButton = new Button("Add Nutrient Filter");
 		nutriQueryButton.setOnAction(e -> {
-			//TODO limit what shows in foodList
+			String[] queryArray = queryRules.getText().split(",");
+			mainFoodList.filterByNutrients(Arrays.asList(queryArray));
 		});
-		nutriQuery.getChildren().addAll(nutriDrop, compDrop, queryAmount, nutriQueryButton);
-		
+		nutriQuery.getChildren().addAll(queryRules, nutriQueryButton);
+
 		queries.getChildren().addAll(nameQuery, orText, nutriQuery);
-		
+
 		GridPane.setConstraints(queries, 0, 3);
-		
+
 		Button clearQuery = new Button("Clear Queries");
-		clearQuery.setAlignment(Pos.BASELINE_CENTER);
+		clearQuery.setOnAction(e -> {
+			nameText.clear();
+			queryRules.clear();
+		});
 		GridPane.setConstraints(clearQuery, 1, 3);
-		
+
 		// ANALYSIS
 		Button analyze = new Button("Analyze Meal");
+		analyze.setOnAction(e -> {
+			analyzeMeal();
+		});
 		GridPane.setConstraints(analyze, 3, 3);
-		
-		TextArea mealAnalysis = new TextArea();
-		GridPane.setConstraints(mealAnalysis, 3, 4);
-		
-		
-		grid.getChildren().addAll(loadFileButton, foodItemInsert, foodListTable, 
-				menuListTable, transferButton, queries, clearQuery, analyze, mealAnalysis);
 
-		Scene mainScene = new Scene(grid);		
+		//		TextArea mealAnalysis = new TextArea();
+		//		GridPane.setConstraints(mealAnalysis, 3, 4);
+
+
+		grid.getChildren().addAll(listMenu, foodItemInsert, foodListTable,
+				menuListTable, transferButton, queries, clearQuery, analyze);
+
+		mainScene = new Scene(grid);
 		mainStage.setScene(mainScene);
 		mainStage.show();
 
 	}
 
+	private void loadFile(){
+		FileChooser fc = new FileChooser();
+		fc.getExtensionFilters().add(new ExtensionFilter("CSV Files", "*.csv", "*.txt"));
+		fc.setTitle("Load Food List");
+		File file = fc.showOpenDialog(mainStage);
+		if (file != null) {
+			mainFoodList.loadFoodItems(file.toString());
+		}
+
+		else {
+			System.out.println("File not found");
+		}
+	}
+
+	private void saveFile() {

```

Comment:
Will you want to hook into the saveFoodItems() method in the FoodData object?

Reply:
saveFoodItems() is used on line 389.

Reply:
And it's functional